### ISSUE TITLE
Infinitely Loop Custom Rule Regex

### ISSUE THREAD

**Issue by EmperiorEric** [realm/SwiftLint#2177]
I've found a custom rule that causes an infinite loop. We're trying to write a custom rule that checks code like this `UserAction(name: "pages_subpagesMigration_view")` to ensure the string doesn't contain any non alphanumeric + underscore characters. I'm not very familiar with regex, so could easily be that as opposed to a SwiftLint bug. But since it goes into an infinite loop somewhere, I'm guessing its a possible fix for SwiftLint perhaps in addition to better regex. This does also get stuck in Xcode's Find and Replace regex as well, so that could mean SwiftLint is off the hook and the regex is solely to blame.

```
  universe_analytics_event_name_format:
    name: "Analytics Event Name Format"
    message: "Firebase restricts event names to alphanumeric and underscore characters."
    severity: error
    regex: 'UserAction\(name: "((\w+_?)+(:\w+)+)*"'
```

I know this bug report deserves more information. Is there a way for me to find swiftlint logs that would highlight the line its getting stuck on?

**Comment by SDGGiesbrecht** [comment]
I do not know where the infinite loop is coming from. (SwiftLint just passes regular expressions off to Foundation, which in turn uses ICU.)

However, @EmperiorEric, your expression does not correspond to its description.

Paste it (`UserAction\(name: "((\w+_?)+(:\w+)+)*"`) into [https://regex101.com](https://regex101.com) and read the English explanation to the right.

Maybe you want something more like this?: `UserAction\(name: "[\w_]*[^"\w_]`

This was the sample source I used:
```swift
UserAction(name: "letters") // valid
UserAction(name: "...invalid...")
```

**Comment by marcelofabri** [comment]
Closing this due to lack of activity.
